implementations/rust/ockam/ockam_entity/src/worker/entity_worker.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+    EntityError::IdentityApiFailed, Identity, IdentityRequest, IdentityRequest::*,
+    IdentityResponse as Res, MaybeContact, ProfileIdentifier, ProfileState,
+};
+
+use async_trait::async_trait;
+use ockam_core::{
+    lib::{result::Result::Ok, HashMap},
+    Result, Routed, Worker,
+};
+use ockam_node::Context;
+use ockam_vault_sync_core::{Vault, VaultSync};
+
+#[derive(Default)]
+pub struct EntityWorker {
+    profiles: HashMap<ProfileIdentifier, ProfileState>,
```

Comment:
Why do we need ProfileState? Why not just Profile?

Reply:
We talked about this a bit..lowest level state vs high level handle, in this case. Maybe we can collapse it further